src/imageset_preview.jsx

```diff
@@ -0,0 +1,108 @@
+import React, { Component } from 'react';
+import { IconImageSet } from './icons'
+
+class ImagesetPreview extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      visibleImages: this.getVisibleImages(this.props.images)
+    }
+  }
+
+  getVisibleImages(images) {
+    let widths = []
+    let hidden = 0
+    images.map((item, i) => {
+      var adjustedWidth = 150 * item.width / item.height
+      widths.push(adjustedWidth)
+      var total = widths.reduce((a,b) => a+b, 0)
+      var margins = widths.length * 10
+      if ((total + margins + 50) > 560) {
+        hidden = hidden + 1
+      }
+    })
+    return widths.length - hidden
+  }
+
+  renderImages(images) {
+    const items = images.slice(0,4).map((item, i) => {
+      var src = item.image ? item.image : item.url
+        if (i < this.state.visibleImages) {
+          return <img
+                  key={'imageset-' + i}
+                  src={src || ''}
+                  className='imageset-preview__image'
+                  style={styles.image} />
+        }
+      }, this)
+    return items
+  }
+
+  render() {
+    const { images } = this.props;
```

Comment:
Not a blocker—but noticing some inconsistent use of semicolons here. Might be worth integrating a [code formatter](https://github.com/prettier/prettier) or [linter](https://standardjs.com/) in here at some point.